### PR TITLE
Fix the problem that the program crashes if the gpuShare Devices is empty

### PR DESCRIPTION
**Pull request by wangyang0616**
fix: [#2738](https://github.com/volcano-sh/volcano/issues/2738)

### CODE REVIEW

File: pkg/scheduler/plugins/predicates/predicates.go [hunk 1]
```diff
@@ -197,12 +197,16 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 			}
 			//predicate gpu sharing
 			for _, val := range api.RegisteredDevices {
-				if nodeInfo.Others[val].(api.Devices).HasDeviceRequest(pod) {
-					err := nodeInfo.Others[val].(api.Devices).Allocate(ssn.KubeClient(), pod)
-					if err != nil {
-						klog.Errorf("AllocateToPod failed %s", err.Error())
-						return
+				if devices, ok := nodeInfo.Others[val].(api.Devices); ok {
+					if devices.HasDeviceRequest(pod) {
```
[the remark this answers is not in the document]
Comment (wangyang0616) [reply pkg/scheduler/plugins/predicates/predicates.go]: done

**Review by william-wang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**